src/main/java/com/movie_collection/gui/controllers/BaseController.java

```diff
@@ -1,35 +1,139 @@
+     * one by one added into the empty scroll pane
+     * if no categories are provided stack pane is filled with Label that notifies the user about not having any category listed yet
+     * LinkedHashMap is added here due to the reason that whenever new Button is added it is automatickly put as the last and keps the order
+     * TODO: Maybe all of the body can be exctracted into separated class since it look so hoooge
+     * @param allCategories list of all categories
+     */
+    private void setCategoriesScrollPane(List<Category> allCategories) {
+        // This code is creating a new Map object that is populated with the key-value pairs of a given Map,
+        //  and then returning it.
+        LinkedHashMap<Button, Button> buttonMap = allCategories.stream()
+                .map(category -> {
+                    Button btn = new Button(category.name().getValue());
+                    Button btn2 = new Button("Delete");
+
+                    // Setting on the action for switching views
+                    btn.setOnAction(event -> {
+                        Parent parent = null;
+                        try {
+                            parent = loadNodesView(ViewType.MOVIES);
+                        } catch (IOException e) {
+                            throw new RuntimeException(e);
+                        }
+                        switchToView(parent); // switches into chosen view
+                    });
+                    btn.setPrefWidth(140);
+
+                    // Setting delete for individual button in order to know which one it is
+                    // if true then we refresh again the scroll pane
+                    btn2.setOnAction(event -> {
+                        var result =  categoryModel.deleteCategory(category.id());
+                        if (result) {
+                            refreshScrollPane();
+                        } else {
+                            throw new RuntimeException("Could not delete category with id: " + category.id());
+                        }
+                    });
+                    btn2.setPrefWidth(50);
+                    return new AbstractMap.SimpleEntry<>(btn, btn2);
+                })
+                // (oldValue, newValue) -> oldValue - a "merge function" that is used to resolve
+                // conflicts when two keys are mapped to the same value. In this case, if there is a conflict,
+                // the old value is kept and the new value is discarded.
+
+                // LinkedHashmap::new  is a function that creates a new, empty Map object. In this case, a LinkedHashMap is created.
+                // collect method is called in order to finish up what we opened at the begging stream
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue,
+                        (oldValue, newValue) -> oldValue, LinkedHashMap::new));
+
```

Comment:
this might not be necessary since SQL query would throw exception before we get to UI I assume and we can resolve it from that error

Reply:
We need to have discussing about error and I will try to explain what we need to be catching ur not only catching sql exceptiona u dont want user to have clue about some exceptions 
Its storing the state in variable and based on that u are doing UI work

Reply:
Lets just stick why we have do requirment and then we can look into it more